### PR TITLE
Added Allure plug in

### PR DESCRIPTION
**Pull request by krishnanuthalapati**
Added Allure plug in

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: jenkins/plugins.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ azure-commons:1.0.5
 azure-credentials:4.0.3
 azure-keyvault:2.1
 azure-vm-agents:1.5.1
+allure:2.29.0
```
Comment (timja) [inline jenkins/plugins.txt:10]: can you give a background to this and why you want it? 

(and please add it in alphabetical order)

Comment (timja) [reply jenkins/plugins.txt:10]: is this the plugin you're after?

if so it's 
```suggestion
allure-jenkins-plugin:2.29.0
```

**Comment by timja** [comment]
Can you add why you want this? and what the plugin will do for you?

**Comment by krishnanuthalapati** [comment]
> Can you add why you want this? and what the plugin will do for you?

1. Allure report is for frontend applications functional tests which is run using multiple workers (parallelisation test).
2. Allure plugin allows us to view functional test reports after each build run.

**Comment by naikajah** [comment]
Hi, @krishnanuthalapati 
General approach on any Jenkins plugin requirement we want the following queries to be resolved: 

Q1: Why and what is the plugin for? What improvements will it have over the existing plugin(s) if there are any existing ones that can be used?
Q2: Has it been tested in the sandbox environment? 
Q3: Does it have a configuration as code functionality?

**Comment by krishnanuthalapati** [comment]
> Hi, @krishnanuthalapati
> General approach on any Jenkins plugin requirement we want the following queries to be resolved:
> 
> Q1: Why and what is the plugin for? What improvements will it have over the existing plugin(s) if there are any existing ones that can be used?

Ans : Current mochawesome reports does not provide embedded screenshots when tests fails in the report, and also it does not support multiple runs using codeceptjs

> Q2: Has it been tested in the sandbox environment?

Ans : It's been tested locally and it works and it won't break any other reporting. As per process definitely we will go first test in sandbox and then will update you. Please can you install Allure plugin in sandbox env.

> Q3: Does it have a configuration as code functionality?

Ans : It does not require specific configuration code in jerkins library as its runtime node dependency embedded in package.json

**Comment by timja** [comment]
Hi @krishnanuthalapati 

I've installed this manually on sandbox-build

Please test and report back here with a working build demonstrating it does what you want

Once done we can review it and merge this (you will need to resolve conflicts, I updated Jenkins on sandbox before installing the plugin)

**Pull request closed without merging**